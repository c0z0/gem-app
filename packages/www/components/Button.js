import styled from 'styled-components'

import { LoadingElipsis } from './Typography'

const Button = styled.button.attrs({ role: 'button' })`
  font-size: 16px;
  color: white;
  background: ${({ theme }) => theme.main};

  padding: 0 2rem;
  line-height: 2.5rem;
  margin: 0;
  border-radius: 7px;
  height: 2.5rem;
  box-shadow: 0px 4px 14px rgba(117, 72, 155, 0.39);
  transition: all 0.2s;
  outline: none;
  border: none;
  cursor: pointer;

  text-transform: capitalize;

  &:hover {
    opacity: 0.9;
    box-shadow: 0px 6px 20px rgba(117, 72, 155, 0.39);
  }

  ${({ secondary, theme }) =>
    secondary &&
    `
    color: ${theme.main};
    background: white;
    box-shadow: 0px 4px 14px rgba(0, 0, 0, 0.1);

    &:hover {
      box-shadow: 0px 6px 20px rgba(0, 0, 0, 0.12);
    }

  `}

${({ small }) =>
  small &&
  `
    height: 2rem;
    line-height: 2rem;

  `}

${({ flat }) =>
  flat &&
  `
    box-shadow: none;
    &:hover {
      box-shadow: none;
    }
  `}

  ${({ loading, disabled }) =>
    (loading || disabled) &&
    `
    cursor: not-allowed;
    color: #CCCCCC;
    box-shadow: none !important;
    opacity: 1 !important;
    background: #EFEFEF;
  `}

`

Button.Elipsis = LoadingElipsis

export default Button
