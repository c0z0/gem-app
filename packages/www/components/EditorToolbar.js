import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Link from 'next/link'

import SubMenu from './SubMenu'
import * as Icons from './Svg'
import { LoadingElipsis } from './Typography'

const Action = styled.button`
  outline: none;
  background: white;
  z-index: 2;
  cursor: pointer;
  padding: 8px 10px;
  font-size: 21px;
  transition: all 0.2s;
  color: #484848;

  display: flex;
  align-items: center;

  &:hover {
    background: #fafafa;
    color: #484848;
  }

  ${({ active }) =>
    active &&
    ` background: red;
    `}

  color: ${({ theme }) => theme.text};
  font-weight: 500;
  border: 1px solid #eaeaea;
  border-right: none;

  &:first-child {
    border-radius: 4px 0 0 4px;
    border-left: 1px solid #eaeaea;
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
    border-right: 1px solid #eaeaea;
  }

  &:only-child {
    border-radius: 4px;
  }


  ${({ divider }) => divider && `border-right: 1px solid #ddd;`}
`

const SaveStatus = styled.span`
  color: #ddd;
  align-self: center;
  margin-left: 8px;
  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex: 1;
    text-align: right;
  }
`

const Controls = styled.div`
  display: flex;
`

const Carret = styled.img.attrs({ src: '/static/carret.svg' })`
  transform: rotate(90deg);
  margin-right: 8px;
  color: #484848;
`

const BackButton = styled.a`
  align-self: center;
  margin-right: 8px;
  display: flex;
  color: #aaa;
  text-decoration: none;
  align-items: center;
  transition: all 0.2s;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex: 1;
  }

  :hover {
    color: #484848;
  }
`

export default function EditorToolbar({
  onToggleMark,
  onToggleBlocks,
  loading,
  unsavedChanges
}) {
  // const currentStyle = editorState.getCurrentInlineStyle()

  return (
    <SubMenu
      active="/notes"
      controls={
        <React.Fragment>
          <Link href="/notes" passHref>
            <BackButton>
              <Carret />
              <span>Back</span>
            </BackButton>
          </Link>
          <Controls>
            <Action onClick={() => onToggleMark('bold')}>
              <Icons.BoldIcon />
            </Action>
            <Action divider onClick={() => onToggleMark('italic')}>
              <Icons.ItalicIcon />
            </Action>
            <Action onClick={() => onToggleBlocks('h2')}>
              <Icons.HeaderBigIcon />
            </Action>
            <Action onClick={() => onToggleBlocks('h3')}>
              <Icons.HeaderSmallIcon />
            </Action>
          </Controls>
          <SaveStatus>
            {!loading ? (
              unsavedChanges ? (
                'Not saved'
              ) : (
                'Saved'
              )
            ) : (
              <React.Fragment>
                Saving
                <LoadingElipsis />
              </React.Fragment>
            )}
          </SaveStatus>
        </React.Fragment>
      }
    />
  )
}

EditorToolbar.propTypes = {
  onToggleMark: PropTypes.func.isRequired,
  onToggleBlocks: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  unsavedChanges: PropTypes.bool.isRequired
}
