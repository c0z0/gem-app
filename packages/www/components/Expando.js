import React, { useEffect, useRef, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  height: ${({ open, openHeight }) => (open ? openHeight : '0px')};
  opacity: ${({ open }) => (open ? '1' : '0')};
  transition: all 0.2s;
  overflow: ${({ open }) => (open ? 'inherit' : 'hidden')};
  @media (${({ theme }) => theme.b.tabletUp}) {
    padding-top: ${({ open }) => (open ? '12px' : '0')};
  }
`

export default function Expando({ children, open }) {
  const contentRef = useRef(null)
  const [heightState, setHeight] = useState('0px')

  useEffect(() => {
    const height = contentRef.current.offsetHeight

    setHeight(`${height}px`)
  }, [children])

  return (
    <Wrapper open={open} openHeight={heightState}>
      <div ref={contentRef}>{children}</div>
    </Wrapper>
  )
}

Expando.defaultProps = {
  children: null
}

Expando.propTypes = {
  open: PropTypes.bool.isRequired,
  children: PropTypes.node
}
