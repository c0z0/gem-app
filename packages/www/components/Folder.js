import React, { useEffect, useState, useRef } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

function outerHeight(el) {
  let height = el.offsetHeight
  const style = getComputedStyle(el)

  height += parseInt(style.marginTop, 10) + parseInt(style.marginBottom, 10)
  return height
}

const Wrapper = styled.div`
  height: ${({ open, openHeight }) => (open ? openHeight : '0px')};
  opacity: ${({ open }) => (open ? '1' : '0')};
  transition: all 0.2s;
  overflow: ${({ open }) => (open ? 'inherit' : 'hidden')};
  @media (${({ theme }) => theme.b.tabletUp}) {
    padding-top: ${({ open }) => (open ? '12px' : '0')};
  }
`

const Carret = styled.img.attrs({ src: '/static/carret.svg' })`
  transform: rotate(${({ flipped }) => (flipped ? '180deg' : '0deg')});

  transition: transform 0.2s;
  margin-right: 10px;
`

const Content = styled.div`
  padding-left: 32px;
`

const FolderWrapper = styled.div`
  &:not(:last-child) {
    border-bottom: 1px solid #eee;
    margin-bottom: 28px;
  }
`

const Title = styled.h4.attrs({ role: 'button' })`
  font-weight: normal;
  padding: 16px 0;
  font-size: 20px;
  cursor: pointer;
  margin: 0;
  display: flex;
  align-items: center;
`

export default function Folder({
  children,
  title,
  onDrop,
  onDragOver,
  onDragLeave
}) {
  const contentRef = useRef(null)
  const [heightState, setHeight] = useState('0px')
  const [visibleState, setVisibleState] = useState(false)

  useEffect(() => {
    const height = outerHeight(contentRef.current)

    setHeight(`${height}px`)
  }, [children])

  return (
    <FolderWrapper
      onDrop={onDrop}
      onDragOver={onDragOver}
      onDragLeave={onDragLeave}
    >
      <Title onClick={() => setVisibleState(!visibleState)}>
        <Carret flipped={visibleState} />
        <span>{title}</span>
      </Title>
      <Wrapper open={visibleState} openHeight={heightState}>
        <Content ref={contentRef}>{children}</Content>
      </Wrapper>
    </FolderWrapper>
  )
}

Folder.defaultProps = {
  children: null,
  onDrop: () => {},
  onDragOver: () => {},
  onDragLeave: () => {}
}

Folder.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string.isRequired,
  onDrop: PropTypes.func,
  onDragOver: PropTypes.func,
  onDragLeave: PropTypes.func
}
