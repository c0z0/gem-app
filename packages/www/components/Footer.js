import React from 'react'
import styled from 'styled-components'

import Container from './Container'
import { Diamond } from './Svg'

const FooterContainer = styled(Container)`
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex-direction: column;
  }
`

const Border = styled.footer`
  background: #fafafa;
  border-top: solid 1px #eaeaea;
  padding: 48px 0;
  color: #cecece;
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
`

const Logo = styled.div`
  display: flex;
  align-items: center;
`

const Title = styled.span`
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: #cecece;
  font-size: 20px;
`

const StyledDiamond = styled(Diamond).attrs({
  alt: 'gem logo'
})`
  width: 40px;
  filter: grayscale(100%);
  opacity: 0.5;
`

const Link = styled.a`
  color: #cecece;
  text-decoration: none;
  transition: all 0.2s;

  &:hover {
    color: #484848;
  }
`

export default function Footer() {
  return (
    <Border>
      <FooterContainer>
        <Logo>
          <StyledDiamond />
          <Title>Gem</Title>
        </Logo>
        <p>
          Made by <Link href="https://cserdean.com">cserdean.com.</Link>
        </p>
      </FooterContainer>
    </Border>
  )
}
