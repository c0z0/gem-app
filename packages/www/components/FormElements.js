import styled from 'styled-components'

export const Input = styled.input`
  width: 275px;
  max-width: 100%;
  padding: 8px;
  font-size: 16px;
  outline: none;
  color: #484848;
  border: 1px solid #ddd;
  border-radius: 6px;
  transition: all 0.2s;
  margin: 1rem 0;

  &::placeholder {
    color: #ddd;
  }

  &:focus {
    border-color: #aaa;
  }

  ${({disabled}) =>
    disabled &&
    `border-color: #ddd; color: #ddd; background: #fafafa; cursor: not-allowed;`}
`

export const Button = styled.button`
  outline: none;
  border: none;
  border-radius: 6px;
  padding: 8px 12px;
  font-size: 16px;
  margin: 4px;
  color: white;
  cursor: pointer;
  background: ${({theme}) => theme.main};
  border: 1px solid transparent;

  ${({disabled}) =>
    disabled &&
    `color: #ddd; border-color: #ddd; background: #fafafa; cursor: not-allowed;`}
`
