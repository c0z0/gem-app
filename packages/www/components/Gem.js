import React, { useState, useEffect, useRef } from 'react'
import styled, { keyframes } from 'styled-components'
import PropTypes from 'prop-types'

import { LoadingElipsis } from './Typography'
import { Dots } from './Svg'

const copyToClipboard = str => {
  const el = document.createElement('textarea')
  el.value = str
  el.setAttribute('readonly', '')
  el.style.position = 'absolute'
  el.style.left = '-9999px'
  document.body.appendChild(el)
  el.select()
  document.execCommand('copy')
  document.body.removeChild(el)
}

const Container = styled.div`
  padding-right: 18px;
  padding-bottom: 28px;
  position: relative;
  color: #ddd;
`

const FakeMargin = styled.div`
  padding: 14px 0;

  &:not(:last-child) ${Container} {
    border-bottom: 1px solid #eee;
  }
`
const Title = styled.a`
  color: ${({ theme }) => theme.main};
  text-decoration: none;
  transition: all 0.2s;
  padding-right: 8px;
  padding-bottom: 8px;

  &:hover {
    text-decoration: underline;
  }
`

const OptionsButton = styled.button`
  background: none;
  border: none;
  outline: none;

  cursor: pointer;
  transition: all 0.2s;
`

const OptionsContainer = styled.div`
  position: absolute;
  top: -4px;
  right: -4px;
  padding: 8px;
`

const Url = styled.a`
  color: #aaa;
  text-decoration: none;
  transition: all 0.2s;

  &:hover {
    text-decoration: underline;
  }
`

export const Tag = styled.span`
  margin: 4px;
  margin-left: 0;
  padding: 4px 8px;
  font-size: 12px;
  cursor: pointer;
  border-radius: 4px;
  background: #eee;
  color: #aaa;
  transition: all 0.2s;

  &:hover {
    color: white;
    background: ${({ theme }) => theme.secondary};
  }
`

const MenuEntry = keyframes`
  0% {
    transform: translateY(-50%);
    opacity: 0; 
  }

  100% {
    transform: translateY(0);
    opacity: 1; 
  }
`

const MenuEntryBottom = keyframes`
  0% {
    transform: translateY(50%);
    opacity: 0; 
  }

  100% {
    transform: translateY(0);
    opacity: 1; 
  }
`

const AlignHorizontal = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`

const MenuBackground = styled.div`
  display: none;

  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: ${({ visible }) => (visible ? '0.2' : '0')};
  pointer-events: ${({ visible }) => (visible ? 'auto' : 'none')};
  transition: all 0.2s;

  background: black;
  z-index: 50;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    display: block;
  }
`

const Menu = styled.ul`
  position: absolute;
  z-index: 1;
  right: 0;
  animation: ${MenuEntry} 0.2s;
  padding: 0;
  min-width: 150px;
  border-radius: 5px;
  background: white;
  margin: 0;
  box-shadow: 0 10px 20px 0 rgba(0, 0, 0, 0.12);

  display: ${({ visible }) => (visible ? 'block' : 'none')};

  @media (${({ theme }) => theme.b.tabletUp}) {
    top: calc(100% + 2px);
  }

  @media (${({ theme }) => theme.b.phoneOnly}) {
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 100;
    box-shadow: none;
    border-radius: 0;
    animation: ${MenuEntryBottom} 0.2s;
  }
`

const MenuItem = styled.li`
  padding: 0 18px;
  font-size: 14px;
  cursor: pointer;
  white-space: nowrap;
  line-height: 40px;
  color: ${({ red }) => (red ? 'red' : '#484848')};

  list-style: none;
  text-align: left;
  border-radius: 5px;
  display: block;
  transition: all 0.2s;

  &:hover {
    background: #fafafa;
  }

  &:not(:last-child) {
    border-radius: 5px 5px 0 0;
    border-bottom: 1px #ddd solid;
  }

  @media (${({ theme }) => theme.b.phoneOnly}) {
    padding: 8px 18px;
  }
`

const Star = styled.img.attrs({ src: '/static/star.svg' })`
  width: 14px;
  margin-right: 8px;
  cursor: pointer;
`

export default function Gem({
  title,
  href,
  displayUrl,
  favorite,
  onDelete,
  id,
  onToggleFavorite,
  onDrag,
  onDragEnd,
  onMoveGem,
  folderId,
  folders
}) {
  const [optionsState, setOptions] = useState(false)
  const [moveMenu, setMoveMenu] = useState(false)
  const [showNewFolder, setShowNewFolder] = useState(false)
  const menuRef = useRef(null)

  // const [newFolderTitle, setNewFolderTitle] = useState('')

  useEffect(() => {
    function listener(e) {
      if (menuRef.current.contains(e.target)) return

      setOptions(false)
      setMoveMenu(false)
    }

    window.addEventListener('click', listener)

    return () => window.removeEventListener('click', listener)
  }, [])

  return (
    <FakeMargin draggable onDrag={onDrag} onDragEnd={onDragEnd}>
      <Container>
        <OptionsContainer id="gem-options-menu" ref={menuRef}>
          <OptionsButton
            onClick={() => {
              setOptions(!optionsState)
              setMoveMenu(false)
            }}
          >
            <Dots />
          </OptionsButton>

          <MenuBackground
            visible={optionsState}
            onClick={() => setOptions(false)}
          />

          <Menu visible={optionsState}>
            {moveMenu ? (
              showNewFolder ? (
                <React.Fragment>
                  <MenuItem onClick={() => setShowNewFolder(false)}>
                    Back
                  </MenuItem>
                </React.Fragment>
              ) : (
                <React.Fragment>
                  <MenuItem onClick={() => setMoveMenu(false)}>Back</MenuItem>
                  <MenuItem
                    onClick={() => setShowNewFolder(true)}
                    style={{ marginBottom: '8px' }}
                  >
                    New folder
                  </MenuItem>
                  {folders.map(f => (
                    <MenuItem
                      key={f.id}
                      onClick={() => onMoveGem({ id, folderId: f.id })}
                    >
                      {f.title}
                    </MenuItem>
                  ))}
                  {folderId && (
                    <MenuItem onClick={() => onMoveGem({ id, folderId: null })}>
                      Remove from folder
                    </MenuItem>
                  )}
                </React.Fragment>
              )
            ) : (
              <React.Fragment>
                <MenuItem
                  onClick={() => {
                    setOptions(false)
                    copyToClipboard(href)
                  }}
                >
                  Copy to clipboard
                </MenuItem>
                <MenuItem
                  onClick={() => {
                    setOptions(false)
                    onToggleFavorite({ id, favorite })
                  }}
                >
                  {favorite ? 'Remove from favorites' : 'Add to favorites'}
                </MenuItem>
                <MenuItem
                  onClick={e => {
                    e.stopPropagation()
                    setMoveMenu(true)
                  }}
                >
                  Move to folder
                </MenuItem>
                <MenuItem
                  red
                  onClick={() => {
                    setOptions(false)
                    onDelete(id)
                  }}
                >
                  Delete
                </MenuItem>
              </React.Fragment>
            )}
          </Menu>
        </OptionsContainer>
        <div>
          <Title href={href}>
            {id === 'optimistic-id' ? (
              <React.Fragment>
                Loading
                <LoadingElipsis />
              </React.Fragment>
            ) : (
              title
            )}
          </Title>{' '}
          <AlignHorizontal>
            {favorite && (
              <Star onClick={() => onToggleFavorite({ id, favorite })} />
            )}
            <Url href={href}>{displayUrl}</Url>
          </AlignHorizontal>
        </div>
      </Container>
    </FakeMargin>
  )
}

Gem.defaultProps = {
  folderId: null,
  onDrag: () => {},
  onDragEnd: () => {}
}

Gem.propTypes = {
  id: PropTypes.string.isRequired,
  folderId: PropTypes.string,
  title: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  displayUrl: PropTypes.string.isRequired,
  favorite: PropTypes.bool.isRequired,
  folders: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      id: PropTypes.string
    })
  ).isRequired,
  onDelete: PropTypes.func.isRequired,
  onDrag: PropTypes.func,
  onDragEnd: PropTypes.func,
  onToggleFavorite: PropTypes.func.isRequired,
  onMoveGem: PropTypes.func.isRequired
}
