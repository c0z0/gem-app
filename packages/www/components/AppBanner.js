import React, { useState, useEffect } from 'react'
import styled, { keyframes } from 'styled-components'
import cookie from 'cookie'

import { P } from './Typography'
import Button from './Button'

const Container = styled.div`
  padding: 16px 16px;
  position: relative;
  display: flex;
  align-items: center;
`

const SlideUp = keyframes`
  from {
    transform: translateY(150%);
    opacity: 0;
  }

  to {
    transform: translateY(0%);
    opacity: 1;
  }
`

const CloseButton = styled.button`
  font-size: 16px;
  color: #484848;
  opacity: 0.5;
  position: absolute;
  top: 2px;
  right: 4px;
  padding: 0;
  transform: rotate(45deg);
`

const Banner = styled.div`
  position: fixed;
  bottom: 16px;
  left: 16px;
  right: 16px;
  z-index: 200;

  box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.1);
  border-radius: 10px;
  background: white;

  animation: ${SlideUp} 0.3s ease-out;
`

export default function AppBanner() {
  const [displayBanner, setDisplayBanner] = useState(false)

  useEffect(() => {
    const bannerDismissed =
      cookie.parse(document.cookie).bannerDismissed || false

    setDisplayBanner(
      navigator.userAgent.toLowerCase().includes('android') && !bannerDismissed
    )
  }, [])

  if (displayBanner)
    return (
      <Banner>
        <Container>
          <P style={{ flex: 1 }}>Download our Android app</P>
          <a href="/download-apk">
            <Button>Download</Button>
          </a>
          <CloseButton
            onClick={() => {
              document.cookie = cookie.serialize('bannerDismissed', true, {
                maxAge: 60 * 60 * 24 * 7 // 1 week
              })

              setDisplayBanner(false)
            }}
          >
            +
          </CloseButton>
        </Container>
      </Banner>
    )
  return null
}
