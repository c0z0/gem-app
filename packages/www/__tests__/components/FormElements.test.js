import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../../lib/theme'
import 'jest-styled-components'

import { Input, Button } from '../../components/FormElements'

afterEach(cleanup)

test('Input matches snapshot', () => {
  const { container, rerender } = render(
    <ThemeProvider theme={theme}>
      <Input />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()

  rerender(
    <ThemeProvider theme={theme}>
      <Input disabled />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('Button matches snapshot', () => {
  const { container, rerender } = render(
    <ThemeProvider theme={theme}>
      <Button />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()

  rerender(
    <ThemeProvider theme={theme}>
      <Button disabled />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})
