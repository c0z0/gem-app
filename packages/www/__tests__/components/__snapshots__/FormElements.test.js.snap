// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
.c0 {
  outline: none;
  border: none;
  border-radius: 6px;
  padding: 8px 12px;
  font-size: 16px;
  margin: 4px;
  color: white;
  cursor: pointer;
  background: #75489B;
  border: 1px solid transparent;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Button matches snapshot 2`] = `
<div>
  <button
    class="sc-bwzfXH hRWzJG"
    disabled=""
  />
</div>
`;

exports[`Input matches snapshot 1`] = `
.c0 {
  width: 320px;
  max-width: 100%;
  padding: 10px;
  font-size: 16px;
  outline: none;
  color: #484848;
  border: 1px solid #ddd;
  border-radius: 7px;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin: 1rem 0;
}

.c0::-webkit-input-placeholder {
  color: #ddd;
}

.c0::-moz-placeholder {
  color: #ddd;
}

.c0:-ms-input-placeholder {
  color: #ddd;
}

.c0::placeholder {
  color: #ddd;
}

.c0:focus {
  border-color: #aaa;
}

<div>
  <input
    class="c0"
  />
</div>
`;

exports[`Input matches snapshot 2`] = `
<div>
  <input
    class="sc-bdVaJa jPmviR"
    disabled=""
  />
</div>
`;
